src/common/reconciler.go

```diff
@@ -93,6 +93,11 @@ func (r *ServiceReconciler) lookupService(registration opslevel_jq_parser.Servic
 		} else if foundService == nil {
 			log.Warn().Msgf("unexpected happened: got service with alias '%s' but the result is nil", alias)
 		} else if foundService.Id == "" {
+			if len(foundService.Aliases) == 1 {
+				// If this happens and there is only 1 alias to check we cannot assume the service doesn't exist
+				// because it seems like our API has a race condition looking up the service
+				return nil, serviceAliasesResult_APIErrorHappened
```

Comment:
@davidbloss i went with this but we could potentially create a new error type that effective does a "lookup retry" that we'd setup # of retires amount so we don't infinite loop.  That way if its a transient thing it would still eventually reconcile in the momemt.

The way it stands now it will just defer reconciliation until the next "event" or whatever the full sync interval the customer has specific is - default 24 hours.  Assuming this lookup failure is transient.